### PR TITLE
ch4/rma: warning fix for incompatible pointer type

### PR DESCRIPTION
**Pull request by minsii**

### CODE REVIEW

**Comment by minsii** [comment]
test:jenkins/ch4/most

File: src/mpid/ch4/shm/posix/posix_win.h [hunk 1]
```diff
@@ -507,7 +507,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_win_free_hook(MPIR_Win * win)
 
         mpi_errno = MPIDI_CH4U_destroy_shm_segment(sizeof(MPL_proc_mutex_t),
                                                    &posix_win->shm_mutex_segment_handle,
-                                                   &posix_win->shm_mutex_ptr);
+                                                   (char **) &posix_win->shm_mutex_ptr);
```
Comment (hajimefu) [inline src/mpid/ch4/shm/posix/posix_win.h]: Does it really have to be `char **`? If pointer arithmetic is not involved, I would prefer `void **` for this kind of types.

Comment (wesbland) [reply src/mpid/ch4/shm/posix/posix_win.h]: Particularly since takes a `void **` anyway.

**Comment by minsii** [comment]
Yes, it makes more sense to use `void *`. The reason I used `char *` was because the internal `MPL_shm` requires `char *` type. After discussion, it seems there is no reason to use `char *` for `MPL_shm`. So I will change it to `void *`, for both `MPL_shm` and `MPIDI_CH4U_destroy_shm_segment`.

**Comment by minsii** [comment]
test:jenkins/ch4/most

**Comment by minsii** [comment]
test:jenkins/ch3/all

**Comment by minsii** [comment]
@hajimefu @raffenet Could you please review this PR ? It contains change in mpl, ch4, and ch3. The ch3/ofi failures are unrelated. I will test MPL_shm/sysv locally.

**Comment by raffenet** [comment]
Looks good to me.

**Comment by hajimefu** [comment]
test:intel:ch4/ofi/sockets/gnu/all/force-am-disabled/debug,default

**Comment by hajimefu** [comment]
test:intel:ch4/ofi/sockets/gnu/all/force-am-disabled/debug,default

**Comment by hajimefu** [comment]
test:intel:ch4/ofi/sockets/gnu/force-am-disabled/all/debug,default

**Comment by raffenet** [comment]
@hajimefu looks like the intel tests are good. can I merge it?

**Comment by minsii** [comment]
Noticed error when use  SystemV version of MPL_shm. Created a ticket #3268 for it.

**Pull request merged**